src/editor/components/modals/file-modal/manage-file-sources/add-new-file-source.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import styled from "react-emotion";
+
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormHelperText from "@material-ui/core/FormHelperText";
+
+import TextInput from "./text-input";
+import { ContainedButton } from "../../../../../shared/components/buttons";
+
+import { addFileSource as addFileSourceAction } from "../../../../actions/file-source-actions";
+import {
+  updateFileSourceInputFilename as updateFileSourceInputFilenameAction,
+  updateFileSourceInputURL as updateFileSourceInputURLAction,
+  updateFileSourceInputUpdateInterval as updateFileSourceInputUpdateIntervalAction
+} from "../../../../actions/file-source-inputs-actions";
+import { FILE_SOURCE_UPDATE_SELECTOR_OPTIONS } from "../../../../state-schemas/state-schema";
+
+const AddNewSourceContainer = styled.div`
+  display: grid;
+  grid-template-columns: auto auto auto auto;
+  align-items: start;
+  grid-column-gap: 40px;
+`;
+
+const AddNewSourceButtonContainer = styled.div`
+  position: relative;
+`;
+
+const AddNewSourceButton = styled(ContainedButton)`
+  justify-self: start;
+  margin: 0;
+`;
+
+const FileSourceStatus = styled.div`
+  min-height: 1em;
+  overflow: hidden;
+  text-align: center;
+  margin-top: 10px;
+  margin-bottom: 10px;
+`;
+
+const FileSourceStatusText = styled.div`
+  transform: translateY(${props => (props.isVisible ? "0" : "1.5em")});
+  transition: 500ms;
+  color: ${props => (props.statusType === "ERROR" ? "red" : "black")};
+`;
+
+const ALLOWED_PROTOCOLS = ["https", "http"];
+
+const hasAllowedProtocol = url => {
+  return ALLOWED_PROTOCOLS.some(protocol => {
+    return url.startsWith(protocol);
+  });
+};
+
+export function AddNewFileSourceUnconnected({
+  filename,
+  url,
+  updateInterval,
+  updateFileSourceInputFilename,
+  updateFileSourceInputURL,
+  updateFileSourceInputUpdateInterval,
+  addFileSource
+}) {
+  const [statusVisible, updateStatusVisibility] = useState(false);
+  const [status, updateStatus] = useState({ type: "NONE", text: "" });
+
+  const handleUpdateIntervalChange = event => {
+    updateFileSourceInputUpdateInterval(event.target.value);
+  };
+  const submitInformation = async () => {
+    updateStatusVisibility(true);
```

Comment:
on further inspection, since it was requested we put all the state in this component in redux, I should probably move the status reporting there as well.